### PR TITLE
Map PDA Program - POIs

### PR DESCRIPTION
**Pull request by DreamySkrell**
image
image
image

### CODE REVIEW

**Review by nauticall — APPROVED** [review]

File: code/controllers/subsystems/initialization/holomap.dm [hunk 1]
```diff
@@ -21,6 +21,9 @@ SUBSYSTEM_DEF(holomap)
 	/// Same as `minimaps_base64`, but does not discriminate between walls and paths.
 	var/list/minimaps_scan_base64 = list()
 
+	/// List of all `/obj/effect/landmark/minimap_poi`.
+	var/list/pois = list()
```
Comment (FluffyGhoster) [inline code/controllers/subsystems/initialization/holomap.dm:25]: Typepath this list if it's supposed to contain only those:

var/list/obj/effect/landmark/minimap_poi/pois

File: code/modules/holomap/poi.dm [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	..()
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/effect/landmark/minimap_poi/LateInitialize()
+	. = ..()
+	SSholomap.pois += src
+
+/obj/effect/landmark/minimap_poi/Destroy()
+	. = ..()
+	SSholomap.pois -= src
```
Comment (FluffyGhoster) [inline code/modules/holomap/poi.dm:18]: Clean up after yourself first, unless there's a compelling reason not to, in Destroy()

File: code/modules/modular_computers/file_system/programs/generic/map.dm [hunk 3]
```diff
@@ -45,6 +45,16 @@
 		list("d"="Hangar", "c"=HOLOMAP_AREACOLOR_HANGAR),
 	)
 
+	data["pois"] = list()
+	for(var/obj/effect/landmark/minimap_poi/poi in SSholomap.pois)
```
Comment (FluffyGhoster) [inline code/modules/modular_computers/file_system/programs/generic/map.dm:49]: If the pois list contains only those, skip the useless typecheck of for:

```suggestion
	for(var/obj/effect/landmark/minimap_poi/poi as anything in SSholomap.pois)
```

File: code/modules/holomap/poi.dm (hunk 2 above)
Comment (DreamySkrell) [reply code/modules/holomap/poi.dm:18]: done

File: code/modules/modular_computers/file_system/programs/generic/map.dm (hunk 3 above)
Comment (DreamySkrell) [reply code/modules/modular_computers/file_system/programs/generic/map.dm:49]: done

File: code/modules/holomap/poi.dm (hunk 2 above)
Comment (DreamySkrell) [reply code/modules/holomap/poi.dm:18]: not sure I understand - do you mean to  mpve the `-= src` above the `. = ..()`?

Comment (FluffyGhoster) [reply code/modules/holomap/poi.dm:18]: Yes

Comment (DreamySkrell) [reply code/modules/holomap/poi.dm:17]: done

**Review by FluffyGhoster — APPROVED** [review]

**Pull request merged**